### ISSUE TITLE
struct.error: unpack requires a buffer of 8 bytes

### ISSUE THREAD

**Issue by ChrisTucker-CNL** [ianare/exif-py#138]
ExifRead 2.3.2
Python 3.6.8

    tags = exifread.process_file(f)

on attached image gives
![126](https://user-images.githubusercontent.com/31433925/104728840-697fe180-572f-11eb-859e-2960f334e8c8.JPG)

(env) [ctucker@development Python]$ python exifread_test.py
Possibly corrupted field ISOSetting in MakerNote IFD
Possibly corrupted field ISOSetting in MakerNote IFD
Possibly corrupted field ISOSetting in MakerNote IFD
Traceback (most recent call last):
  File "exifread_test.py", line 22, in <module>
    tags = exifread.process_file(f)
  File "/home/ctucker/Python/env/lib64/python3.6/site-packages/exifread/__init__.py", line 293, in process_file
    hdr.decode_maker_note()
  File "/home/ctucker/Python/env/lib64/python3.6/site-packages/exifread/classes.py", line 450, in decode_maker_note
    tag_dict=makernote.nikon.TAGS_NEW, relative=1)
  File "/home/ctucker/Python/env/lib64/python3.6/site-packages/exifread/classes.py", line 318, in dump_ifd
    self._process_tag(ifd, ifd_name, tag_entry, entry, tag, tag_name, relative, stop_tag)
  File "/home/ctucker/Python/env/lib64/python3.6/site-packages/exifread/classes.py", line 243, in _process_tag
    values = self._process_field(tag_name, count, field_type, type_length, offset)
  File "/home/ctucker/Python/env/lib64/python3.6/site-packages/exifread/classes.py", line 166, in _process_field
    value = struct.unpack(unpack_format, byte_str)
struct.error: unpack requires a buffer of 8 bytes

**Comment by logofun** [comment]
the same problem
![image](https://user-images.githubusercontent.com/22115036/112659068-00cf8680-8e8f-11eb-926c-489611a11b81.png)

**Comment by Harmouch101** [comment]
That's weird!
I tried @ChrisTucker-CNL image while debugging the code. As the terminal shows "Possibly corrupted field ISOSetting in MakerNote IFD", it seems like some headers of the image were corrupted. However, the struct module is trying to unpack the following bytes-like object `buf = b'\x1a\xfa\x05e'` which is 4 bytes long with `fmt = '>I'` which indicates that unpack was expecting 4 bytes. So, I decided to double-check the result in a separate terminal which worked fine:

```python
>>> import struct
>>> len(b'\x1a\xfa\x05e')
4
>>> struct.unpack('>I', b'\x1a\xfa\x05e')
(452592997,)
```
I am not sure why the error message says that unpack was expecting 8 bytes. that seems odd! But, I will check the traceback and make the project able to handle corrupted files if you don't care about tags that have no values or corrupted ones.
